openml/datasets/functions.py

```diff
@@ -42,7 +48,8 @@ def _list_cached_datasets():
 
             if "dataset.arff" in dataset_directory_content and \
                     "description.xml" in dataset_directory_content:
-                datasets.append(did)
+                if did not in datasets:
```

Comment:
how could it already be there? If that was the case, please add a non-regression test checking that after this change, no duplicates appear.